examples/platforms/qpg6095/platform.c

```diff
@@ -0,0 +1,84 @@
+/*
+ *  Copyright (c) 2019, The OpenThread Authors.
+ *  All rights reserved.
+ *
+ *  Redistribution and use in source and binary forms, with or without
+ *  modification, are permitted provided that the following conditions are met:
+ *  1. Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *  2. Redistributions in binary form must reproduce the above copyright
+ *     notice, this list of conditions and the following disclaimer in the
+ *     documentation and/or other materials provided with the distribution.
+ *  3. Neither the name of the copyright holder nor the
+ *     names of its contributors may be used to endorse or promote products
+ *     derived from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ *  AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ *  IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ *  ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ *  LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ *  CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ *  SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ *  INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ *  CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ *  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ */
+
+/**
+ * @file
+ * @brief
+ *   This file includes the platform-specific initializers.
+ */
+
+#include "alarm_qorvo.h"
+#include "platform_qorvo.h"
+#include "radio_qorvo.h"
+#include "random_qorvo.h"
+#include "uart_qorvo.h"
+#include <openthread/tasklet.h>
+#include <openthread/platform/uart.h>
+
+#include "stdio.h"
+#include "stdlib.h"
+
+static otInstance *localInstance = NULL;
+
+bool qorvoPlatGotoSleepCheck(void)
```

Comment:
This seems inconsistent with the function pointer type declaration. Should this return `uint8_t`? Or should the function pointer return `bool`?

Compile error when using arm-none-eabi-gcc version 8.2.1 20181213:
```
examples/platforms/qpg6095/platform.c: In function 'otSysInit':
examples/platforms/qpg6095/platform.c:64:19: error: cast between incompatible function types from '_Bool (*)(void)' to 'uint8_t (*)(void)' {aka 'unsigned char (*)(void)'} [-Werror=cast-function-type]
     qorvoPlatInit((qorvoPlatGotoSleepCheckCallback_t)&qorvoPlatGotoSleepCheck);
```